### PR TITLE
update bevy 0.11 - fix compile errors

### PR DESCRIPTION
**Pull request by johnny-smitherson**
see issue  #541

compiles then panics

feel free to take over the upgrade, i usually only login on weekends

### CODE REVIEW

File: src_testbed/lines/render_dim.rs [hunk 1]
```diff
@@ -54,17 +55,17 @@ pub mod r3d {
...
+                shader_defs.push("DEPTH_TEST_ENABLED".to_string().into());
             }
 
             let vertex_buffer_layout = layout.get_layout(&[
                 Mesh::ATTRIBUTE_POSITION.at_shader_location(0),
                 Mesh::ATTRIBUTE_COLOR.at_shader_location(1),
             ])?;
             let (label, blend, depth_write_enabled);
-            if key.contains(MeshPipelineKey::TRANSPARENT_MAIN_PASS) {
+            if key.contains(MeshPipelineKey::BLEND_ALPHA) {
```
Comment (johnny-smitherson) [inline src_testbed/lines/render_dim.rs:68]: is this correct? i don't know how this system works or what the enums mean. this is also not mentioned in the bevy upgrade notes 0.9-0.10-0.11

**Comment by Ughuuu** [comment]
So does this fix the testbed examples to not freeze also? What is holding it back from getting merged?

**Comment by Ughuuu** [comment]
Tried it and it does fix the testbed. Nice. Hope it gets merged soon. Right now seems the build is failing because of some unused variables.

**Comment by johnny-smitherson** [comment]
ah, unused warnings - i can fix those sure thing 

@sebcrozet thanks for the push - I didn't connect the new Gizmos API being the replacement for Debug Lines, even through I used both

**Comment by johnny-smitherson** [comment]
- I changed the version from 0.17 to 0.18 in some build files to verify in my external app that I run the modified version; if needed can remove those
- I found a bug in the 3D variant about mouse selecting/picking of model

Red X is where the mouse is hovered, actual brick highlighted is flipped for mouse Y coord, mouse X coord is OK.

I don't know if this was added by this patch or was already in the library, will recompile the master branch in a few minutes

mouse1Capture
mouse2Capture

**Comment by sebcrozet** [comment]
Thanks @johnny-smitherson!

**Pull request merged**